### PR TITLE
add makerlab mega controller pin configs

### PR DESCRIPTION
**Pull request by AndySze**
add support for mega controller board

### CODE REVIEW

**Comment by AnHardt** [comment]
Ah. Much better now.
The indentation looks a bit uneven. Did you use tabs? (http://www.marlinfirmware.org/index.php/Coding_Standards)
What is MINIPANEL? Will this follow in a second PR?

File: Marlin/pins_MEGACONTROLLER.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+     //not connected to a pin
+     #define SDCARDDETECT 49
+ #endif //Minipanel
+
+
+   //encoder rotation values
+ #define encrot0 0
+ #define encrot1 2
+ #define encrot2 3
+ #define encrot3 1
```
Comment (thinkyhead) [inline Marlin/pins_MEGACONTROLLER.h]: These 4 lines are no longer needed.

File: Marlin/pins_MEGACONTROLLER.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ * Mega controller pin assignments
+ */
+
+#ifndef __AVR_ATmega2560__
+  #error Oops!  Make sure you have 'Arduino Mega' selected from the 'Tools -> Boards' menu.
+#endif
+
+#if EXTRUDERS > 3
+  #error RUMBA supports up to 3 extruders. Comment this line to keep going.
+#endif
```
Comment (thinkyhead) [inline Marlin/pins_MEGACONTROLLER.h:11]: This error may not apply to Mega, and mentions RUMBA.

File: Marlin/pins_MEGACONTROLLER.h [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ #define X_MAX_PIN 42 //2 //Max endstops default to disabled "-1", set to commented value to enable.
+
+ #define Y_STEP_PIN 65 // A11
+ #define Y_DIR_PIN 66 // A12
+ #define Y_ENABLE_PIN 64//A10
+#ifdef DELTA
+ #define Y_MIN_PIN -1
+#else
+ #define Y_MIN_PIN 38
+#endif //DELTA
```
Comment (thinkyhead) [inline Marlin/pins_MEGACONTROLLER.h]: You don't need to test `DELTA` in your pins file. A `DELTA` will have no min endstops, and the lack of min endstops will cause these pins to be set to -1 later in `pins.h`. So, just set this to 38 and `X_MIN_PIN` to 43.

**Comment by AndySze** [comment]
@thinkyhead thanks, will fix those problems.

File: Marlin/pins_MEGACONTROLLER.h [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    #endif
+
+    #if NUM_SERVOS > 2
+      #define SERVO2_PIN         32
+    #endif
+
+    #if NUM_SERVOS > 3
+      #define SERVO3_PIN         33
+    #endif
+  #endif
```
Comment (thinkyhead) [inline Marlin/pins_MEGACONTROLLER.h]: Blocks like these are getting formatted like this now…

``` cpp
 #ifdef NUM_SERVOS
   #define SERVO0_PIN       30
   #if NUM_SERVOS > 1
     #define SERVO1_PIN     31
     #if NUM_SERVOS > 2
       #define SERVO2_PIN   32
       #if NUM_SERVOS > 3
         #define SERVO3_PIN 33
       #endif
     #endif
   #endif
 #endif
```

Comment (AndySze) [inline Marlin/pins_MEGACONTROLLER.h]: thanks @thinkyhead ,got it.

**Pull request merged**